src/main/java/com/celements/navigation/NavigationConfig.java

```diff
@@ -0,0 +1,221 @@
+      @Nullable Integer toHierarchyLevel, @Nullable Integer showInactiveToLevel,
+      @Nullable String menuPart, @Nullable String dataType, @Nullable SpaceReference nodeSpaceRef,
+      @Nullable String layoutType, @Nullable Integer nrOfItemsPerPage,
+      @Nullable String presentationTypeHint, @Nullable String cmCssClass) {
+    this(true, configName, fromHierarchyLevel, toHierarchyLevel, showInactiveToLevel, menuPart,
+        dataType, nodeSpaceRef, layoutType, nrOfItemsPerPage, presentationTypeHint, cmCssClass);
+  }
+
+  private NavigationConfig(boolean enabled, String configName, Integer fromHierarchyLevel,
+      Integer toHierarchyLevel, Integer showInactiveToLevel, String menuPart, String dataType,
+      SpaceReference nodeSpaceRef, String layoutType, Integer nrOfItemsPerPage,
+      String presentationTypeHint, String cmCssClass) {
+    this.enabled = enabled;
+    SpaceReference theNodeSpaceRef = (nodeSpaceRef != null) ? new SpaceReference(
+        nodeSpaceRef.clone()) : null;
+    this.origValues = new ConfigValues(Strings.emptyToNull(configName), fromHierarchyLevel,
+        toHierarchyLevel, showInactiveToLevel, Strings.emptyToNull(menuPart), Strings.emptyToNull(
+            dataType), theNodeSpaceRef, Strings.emptyToNull(layoutType), nrOfItemsPerPage,
+        Strings.emptyToNull(presentationTypeHint), Strings.emptyToNull(cmCssClass));
+    String theConfigName = Strings.nullToEmpty(origValues.configName);
+    int theFromHierarchyLevel = Math.max(MoreObjects.firstNonNull(origValues.fromHierarchyLevel,
+        DEFAULT_MIN_LEVEL), DEFAULT_MIN_LEVEL);
+    Integer theToHierarchyLevel = MoreObjects.firstNonNull(origValues.toHierarchyLevel,
+        DEFAULT_MAX_LEVEL);
+    Integer theShowInactiveToLevel = MoreObjects.firstNonNull(origValues.showInactiveToLevel,
+        DEFAULT_MIN_LEVEL - 1);
+    String theMenuPart = Strings.nullToEmpty(origValues.menuPart);
+    String theDataType = MoreObjects.firstNonNull(origValues.dataType, PAGE_MENU_DATA_TYPE);
+    String theLayoutType = MoreObjects.firstNonNull(origValues.layoutType, LIST_LAYOUT_TYPE);
+    int theNrOfItemsPerPage = Math.max(MoreObjects.firstNonNull(origValues.nrOfItemsPerPage,
+        UNLIMITED_ITEMS_PER_PAGE), UNLIMITED_ITEMS_PER_PAGE);
+    String thePresentationTypeHint = MoreObjects.firstNonNull(origValues.presentationTypeHint,
+        "default");
+    String theCmCssClass = Strings.nullToEmpty(origValues.cmCssClass);
+    this.ownValues = new ConfigValues(theConfigName, theFromHierarchyLevel, theToHierarchyLevel,
+        theShowInactiveToLevel, theMenuPart, theDataType, origValues.nodeSpaceRef, theLayoutType,
+        theNrOfItemsPerPage, thePresentationTypeHint, theCmCssClass);
+  }
+
+  @Nullable
+  final private <T> T firstNotNullOrNull(@Nullable T firstObj, @Nullable T secondObj) {
+    if (firstObj != null) {
+      return firstObj;
+    } else {
+      return secondObj;
+    }
+  }
```

Comment:
This method can be replaced with `org.apache.commons.lang3.ObjectUtils.firstNonNull(T...)`